classes/Form/ConfigurationForm.php

```diff
@@ -0,0 +1,201 @@
+                            'value' => 0,
+                            'label' => $this->module->l('Disabled'),
+                        ], ],
+                ],
+                [
+                    'type' => 'switch',
+                    'label' => $this->module->l('Anonymize IP'),
+                    'name' => 'GA_ANONYMIZE_ENABLED',
+                    'hint' => $this->module->l('Use this option to anonymize the visitor’s IP to comply with data privacy laws in some countries'),
+                    'values' => [
+                        [
+                            'id' => 'ga_anonymize_enabled',
+                            'value' => 1,
+                            'label' => $this->module->l('Enabled'),
+                        ],
+                        [
+                            'id' => 'ga_anonymize_disabled',
+                            'value' => 0,
+                            'label' => $this->module->l('Disabled'),
+                        ],
+                    ],
+                ],
+            ],
+            'submit' => [
+                'title' => $this->module->l('Save'),
+            ],
+        ];
+
+        if ($is_multistore_active) {
+            $fields_form[0]['form']['input'][] = [
+                'type' => 'switch',
+                'label' => $this->module->l('Enable Cross-Domain tracking'),
+                'name' => 'GA_CROSSDOMAIN_ENABLED',
+                'values' => [
+                    [
+                        'id' => 'ga_crossdomain_enabled',
+                        'value' => 1,
+                        'label' => $this->module->l('Enabled'),
+                    ],
+                    [
+                        'id' => 'ga_crossdomain_disabled',
+                        'value' => 0,
+                         'label' => $this->module->l('Disabled'),
+                    ],
+                ],
+            ];
+        }
+
+        // Load current value
+        $helper->fields_value['GA_ACCOUNT_ID'] = Configuration::get('GA_ACCOUNT_ID');
+        $helper->fields_value['GA_USERID_ENABLED'] = Configuration::get('GA_USERID_ENABLED');
+        $helper->fields_value['GA_CROSSDOMAIN_ENABLED'] = Configuration::get('GA_CROSSDOMAIN_ENABLED');
+        $helper->fields_value['GA_ANONYMIZE_ENABLED'] = Configuration::get('GA_ANONYMIZE_ENABLED');
+
+        return $helper->generateForm($fields_form);
+    }
+
+    /**
+     * treat the form datas if submited
+     *
+     * @return string
+     */
+    public function treat()
+    {
+        $treatmentResult = '';
+        $gaAccountId = Tools::getValue('GA_ACCOUNT_ID');
+        $gaUserIdEnabled = Tools::getValue('GA_USERID_ENABLED');
+        $gaCrossdomainEnabled = Tools::getValue('GA_CROSSDOMAIN_ENABLED');
+        $gaAnonymizeEnabled = Tools::getValue('GA_ANONYMIZE_ENABLED');
+
+        if (!empty($gaAccountId)) {
+            Configuration::updateValue('GA_ACCOUNT_ID', $gaAccountId);
```

Comment:
Would it be useful to add some validation here ? I guess it's supposed to be an int

Reply:
This is not an int. It's format is `UA-XXXXXXXXX-X`